pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -329,6 +329,24 @@ static Py_ssize_t get_attr_length(PyObject *obj, char *attr) {
     return ret;
 }
 
+npy_int64 get_long_attr(PyObject *o, const char *attr) {
+  npy_int64 long_val;
+  PyObject *value = PyObject_GetAttrString(o, attr);
+  long_val = (PyLong_Check(value) ?
+              PyLong_AsLongLong(value) : PyInt_AS_LONG(value));
+  Py_DECREF(value);
+  return long_val;
+}
+
+npy_float64 total_seconds(PyObject *td) {
```

Comment:
> look in the CPython source code

My attempts so far have resulted in test failures and segfaults.  Having me spend more time futzing with C benefits no one.  If you don't like the PR as it is, feel free to close.